python/orion_finance_sdk_py/contracts.py

```
@@ -244,6 +244,48 @@ def _resolve_block(self, value: datetime | int) -> int:
+        Ensures the end block is included when the last daily step did not land
+        on it. Public RPCs are rate-limited - use a dedicated ``RPC_URL`` for
+        long series.
+        """
+        start_block = self._resolve_block(start)
+        end_block = (
+            self.w3.eth.block_number if end is None else self._resolve_block(end)
+        )
+        if end_block < start_block:
+            raise ValueError(
+                f"end block ({end_block}) is before start block ({start_block})"
+            )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)